tests/cases/compiler/newMap.ts

```diff
@@ -1,2 +1,3 @@
 // @lib: es6
 new Map<string>();
+new WeakMap<object>();

```

Comment:
why do we need to support `new WeakMap<object>()` instead of `new WeakMap<object, object>()`? If only the latter were supported, we wouldn't need to add the type parameter defaults in this PR, would we?

Reply:
Removing support for this would be a breaking change, as the other constructor has type parameter defaults:
https://github.com/microsoft/TypeScript/blob/b2c6a56e38bc8cca1573aeadf8c2bda74a67e8e9/src/lib/es2015.collection.d.ts#L66